### PR TITLE
Add expect_table_column_count_to_equal and _to_be_between

### LINKED ISSUE #669: Feature request for column counts (e.g. expect_table_column_count_to_equal)
As of 0.7.7 there are two methods to check the row counts (`expect_table_row_count_to_be_between ` and `expect_table_row_count_to_equal`), but there are none to check the column counts. Yes, there are more strict checks for columns like `expect_column_to_exist ` and `expect_table_columns_to_match_ordered_list()`. However, sometimes you just need to check the shape and not existence by name. 

This feature request is for two column count expectations to be created that are analogous to the ones for row counts.

### PR DESCRIPTION
**Pull request by elsander**
This PR adds table column expectations to match those for rows (`expect_table_column_count_to_equal` and `expect_table_column_count_to_be_between`). I kept the interface and test suite as close to the row-based versions as possible. I also added the new expectations to the jupyter UX.

It looks like there's a bunch of code related to profiling/rendering; I believe the profile already provides high-level column information, so I didn't add anything there. I'm not sure I fully understand that part of the codebase, so I'm happy for any suggestions or guidance on changes that need to be made there to bring these expectations to feature parity with others.

Just a flag that I'm currently merging into `v0.8.x` since it seems that this is the branch under active development. I can cherry pick these changes over and merge into `develop` if that would be better.

(also note that the CLA check will fail right now; I should be able to address that tomorrow, but thought I could get code review started now)

Closes #669.

### CODE REVIEW

File: great_expectations/dataset/dataset.py [hunk 1]
```diff
@@ -486,6 +491,160 @@ def expect_table_columns_to_match_ordered_list(
...
+
+        Keyword Args:
+            min_value (int or None): \
+                The minimum number of columns, inclusive unless strict_min=True.
+            max_value (int or None): \
+                The maximum number of columns, inclusive unless strict_max=True.
+            strict_min (boolean):
+                If True, the table column count must be strictly larger than min_value.
+            strict_max (boolean):
+                If True, the table column count be strictly smaller than max_value.
```
Comment (jcampbell) [inline great_expectations/dataset/dataset.py]: I don't think this expectation should include strict_min or strict_max since column counts must be integers

**Review by jcampbell — CHANGES_REQUESTED** [review]
This looks great. Only minor request is to drop strict_min and strict_max support (since I think of those as useful only in the case of expectations over truly continuously-valued data)

File: great_expectations/dataset/dataset.py (hunk 1 above)
Comment (elsander) [reply great_expectations/dataset/dataset.py]: These arguments are also present for `expect_table_row_count_to_be_between`. I agree that they're unnecessary. Want me to remove the arguments from both?

Comment (jcampbell) [reply great_expectations/dataset/dataset.py]: Yes please -- good catch.

Comment (elsander) [reply great_expectations/dataset/dataset.py]: I found a couple other integer-based expectations with these args, so I removed it in all of those cases.

**Review by jcampbell — APPROVED** [review]

**Pull request merged**